server/scripts/import-tests/index.js

```diff
@@ -0,0 +1,255 @@
+/* eslint no-console: 0 */
+
+const path = require('path');
+const nodegit = require('nodegit');
+const { Client } = require('pg');
+const fse = require('fs-extra');
+const np = require('node-html-parser');
+
+const args = require('minimist')(process.argv.slice(2), {
+    alias: {
+        h: 'help'
+    }
+});
+
+if (args.help) {
+    console.log(`
+Default use:
+  No arguments:
+    Will fetch most recent aria-at tests to update database.
+  Arguments:
+    -h, --help
+       Show this message.
+`);
+    process.exit();
+}
+
+const client = new Client();
+
+const ariaAtRepo = 'https://github.com/w3c/aria-at.git';
+const tmpDirectory = path.resolve(__dirname, 'tmp');
+const testDirectory = path.resolve(tmpDirectory, 'tests');
+const supportFile = path.resolve(testDirectory, 'support.json');
+
+const ariaat = {
+    /**
+     * Get all tests in the master HEAD commit for the detaul repository
+     */
+    async getMostRecentTests() {
+        await client.connect();
+
+        fse.ensureDirSync(tmpDirectory);
+        let repo = await nodegit.Clone(ariaAtRepo, tmpDirectory, {});
+        console.log(
+            'Cloned ' + path.basename(ariaAtRepo) + ' to ' + repo.workdir()
+        );
+
+        let latestCommit = fse
+            .readFileSync(
+                path.join(tmpDirectory, '.git', 'refs', 'heads', 'master'),
+                'utf8'
+            )
+            .trim();
+        let commit = await nodegit.Commit.lookup(repo, latestCommit);
+        let commitDate = commit.date();
+
+        const testVersionID = await this.upsertTestVersion(
+            ariaAtRepo,
+            latestCommit,
+            commitDate
+        );
+
+        const support = JSON.parse(fse.readFileSync(supportFile));
+        for (let at of support.ats) {
+            const atID = await this.upsertAt(at.name);
+            await this.upsertAtKey(at.key, atID, testVersionID);
+        }
+
+        const exampleName = {};
+        for (let example of support.examples) {
+            exampleName[example.directory] = example.name;
+        }
+
+        let exampleDirs = fse.readdirSync(testDirectory);
+        for (let i = 0; i < exampleDirs.length; i++) {
+            const exampleDir = exampleDirs[i];
+            const subDirFullPath = path.join(testDirectory, exampleDir);
+            const stat = fse.statSync(subDirFullPath);
+
+            if (stat.isDirectory() && exampleDir !== 'resources') {
+                const exampleID = await this.upsertAPGExample(
+                    exampleDir,
+                    exampleName[exampleDir],
```

Comment:
In the `tests/` directory in https://github.com/w3c/aria-at/tree/master/tests, there are more examples in the directory than in `support.json`. When `exampleName[exampleDir]` doesn't exist, that data will be null in the table. Is that ok?

Reply:
yup that right, good catch. We are importing all tests whether or not there is a entry in support.json. The application will fall back to showing the directory name if it doesn't have a fancy name for the APG example from the support.json file. I figured that admins configuring test runs might want to see all tests regardless of what is in support.json.